core/Updater/Migration/Db/DropColumns.php

```diff
@@ -0,0 +1,43 @@
+<?php
+/**
+ * Matomo - free/libre analytics platform
+ *
+ * @link https://matomo.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+namespace Piwik\Updater\Migration\Db;
+
+use Piwik\Db;
+use Piwik\DbHelper;
+
+/**
+ * @see Factory::dropColumn()
+ * @ignore
+ */
+class DropColumns extends Sql
+{
+    public function __construct($tableName, $columnNames)
+    {
+        $table = DbHelper::getTableColumns($tableName);
+
+        // we need to remove all not existing columns. Otherwise if only one of the columns doesn't exist, all of
+        // the columns wouldn't be removed
+        $columnNames = array_filter($columnNames, function ($columnName) use ($table) {
+            return isset($table[$columnName]);
+        });
+
+        if (empty($columnNames)) {
+            parent::__construct('', static::ERROR_CODE_COLUMN_NOT_EXISTS);
```

Comment:
could alternatively use ```alter table `table_name`;``` but I reckon it's less confusing when no SQL query is shown.

Reply:
makes sense to show no sql in this case.